### ISSUE TITLE
Make Kivy apps accessible, add `kivy.core.accessibility` abstraction for accessibility providers

### ISSUE THREAD

**Issue by misl6** [kivy/kivy#8596]
In the last few days, I had the chance to chat with one of the https://github.com/AccessKit/accesskit core developers, regarding accessibility in Kivy.

For the last decade, Kivy has evolved quite a lot but never officially supported accessibility features (even if users have succeded in adding such features by themself).

In our vision to make Kivy a solid platform to build Apps for everyone, introducing accessibility features and making Kivy apps accessible by default has become increasingly important.

Kivy `3.0.0` is the right moment to break things and make the appropriate changes to allow accessibility tools to do their job, when they need to.

Even if we will work closely with `AccessKit` team (as they already have a POC, and offered to write an implementation for their tool in Kivy), Kivy will have an abstraction that will make it possible for other accessibility tools to add their implementation in future.

The plan:
- [ ] Expose a method on `kivy.core.window` to get the window handle
- [ ] Create an event/callback on `kivy.core.window` so accessibility implementations can know when it's the right moment to attach to the Window
- [ ] Write an abstraction for `kivy.core.accessibility`, so accessibility tools have a proper API to work with
- [ ] Introduce accessibility-related properties on Widgets

**Comment by clayote** [comment]
[Here's the proof of concept](https://github.com/DataTriny/kivy/tree/accesskit-demo) thanks to @DataTriny.

**Comment by misl6** [comment]
@DataTriny and @clayote :

I've just finished the first draft of all the tasks linked to this issue.
With no rush, when you have some time, can you please proofread it and share your thoughts?

(mainly https://github.com/kivy/kivy/issues/8599)

**Comment by lilmike** [comment]
Has there been any more thoughts on this? I find myself wanting to use a kivy app and not able to.

-Michael.

**Comment by clayote** [comment]
Well, I did convert @DataTriny 's proof of concept to sdl3...I think? It doesn't make my screen reader say anything, but I'm on Linux, and the proof of concept was written on Windows. Maybe someone on Windows could verify.

https://github.com/clayote/kivy/tree/accesskit

**Comment by clayote** [comment]
No, I'm pretty sure my port is broken, too, because AccessKit has changed its API, and the Python bindings haven't caught up yet.

**Comment by clayote** [comment]
Opened the draft PR #9169

**Comment by DataTriny** [comment]
Thanks @clayote for your interest in this. I've just released accesskit 0.7 on pypi which contain the latest changes from the Rust crates.

**Comment by clayote** [comment]
I have Kivy interacting with my screen reader on Linux.

To maximize the number of existing apps that will work, I'd like to retrofit the existing widgets and such. I'm considering what that implies about the concept of focus, when an app does not have any widgets with `FocusBehavior`.

Perhaps `MotionEvent.grab_current` would be a suitable proxy? It's used to determine which widget gets the motion event sent to it, when there's a need to choose--but sometimes there isn't, and every widget in `MotionEvent.grab_list` gets the event.
